pyproject.toml

```diff
@@ -1,22 +1,45 @@
 [project]
 name = "dc_django_utils"
-version = "6.1.0"
+dynamic = ["version"]
 description = "This is a package to provide some common utilities for Django projects at Democracy Club."
 readme = "README.md"
 authors = [{name = "Sym Roe"}, {name = "Virginia Dooley"}]
 
 dependencies = [
-	"django>=3.2,<4.3",
-	"django-pipeline==3.0.0",
-	"markdown",
-	"django-localflavor==4.0",
-	"setuptools==69.2.0",
-	"whitenoise",
-	"pysass",
-	"jsmin<3.1",
+    "django>=3.2,<4.3",
+    "django-pipeline>=3.0.0",
+    "markdown",
+    "django-localflavor>=4.0",
+    "setuptools>=69.2.0",
+    "whitenoise",
+    "pysass",
+    "jsmin<3.1",
+    "pytidylib",
 ]
 
+[project.optional-dependencies]
+dev = [
+    "black==22.10.0",
+    "curlylint==0.13.1",
+    "isort==5.13.2",
+    "djhtml==3.0.6",
+    "ruff==0.6.1",
+    "pre-commit>=3.0.0",
+    "ipdb==0.13.13",
+    "pytest==8.3.2",
+    "pytest-django==4.8.0",
+    "pytest-mock==3.14.0",
+    "pytest-flakes==4.0.5",
+    "pytest-ruff==0.4.1",
+    "dc-design-system @ https://github.com/DemocracyClub/design-system/archive/refs/tags/0.5.0.tar.gz",
+]
+ci = [
+    "tox==4.18.0"
+]
```

Comment:
Here I have just taken everything from requirements that is not in the main requirements, duplicated it and pinned to the latest version. The one exception to this is `pre-commit`. I've made this a range because we're still running the tests on python 3.8 and the latest version is not compatible with 3.8. This is one of the reasons why it is sometimes useful to be a bit looser with the dev requirements on packages. You generally run the tests on a bunch of different versions of python so it can be useful for the python 3.8 build to use slightly different versions of your dev packages than your python 3.12 build, which is harder in lockfile-world.

There are several Yaks that I am definitively not going to shave in this PR (but I'll come back to in follow up PRs):

1. Bin off black and isort - just use ruff
2. Bin off curlylint. Just use djhtml
3. Stop running tests on old python versions that none of our apps use any more (e.g: python 3.8)

but I'm going to stop pulling threads in this PR. There has already been quite a lot of scope creep.